components/calendar/src/chinese_based.rs

```diff
@@ -52,15 +52,60 @@ pub(crate) trait ChineseBased: CalendarArithmetic + Sized {
     /// reflect traditional methods of year-tracking or eras, since Chinese-based calendars
     /// may not track years ordinally in the same way many western calendars do.
     const EPOCH: RataDie;
+
+    /// Get the compiled const data for a ChineseBased calendar; can return `None` if the given year
+    /// does not correspond to any compiled data.
+    fn get_compiled_data_for_year(year: i32) -> Option<ChineseBasedCompiledData>;
 }
 
 /// Chinese-based calendars define DateInner as a calendar-specific struct wrapping ChineseBasedDateInner.
 #[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
 pub(crate) struct ChineseBasedDateInner<C: ChineseBased>(
     pub(crate) ArithmeticDate<C>,
-    pub(crate) ChineseBasedCache,
+    pub(crate) ChineseBasedYearInfo,
 );
 
+/// A `ChineseBasedDateInner` has additional information about the year corresponding to the Inner;
+/// if there is available data for that year, the ChineseBasedYearInfo will be in the form of `Data`,
+/// with a `ChineseBasedCompiledData` struct which contains more information; otherwise, a `Cache`
+/// with a `ChineseBasedCache`, which contains less information, but is faster to compute.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub(crate) enum ChineseBasedYearInfo {
+    Cache(ChineseBasedCache),
+    Data(ChineseBasedCompiledData),
+}
+
+impl ChineseBasedYearInfo {
+    pub(crate) fn get_new_year(&self) -> RataDie {
+        match self {
+            Self::Cache(cache) => cache.new_year,
+            Self::Data(data) => data.new_year,
+        }
+    }
+
+    pub(crate) fn get_next_new_year(&self) -> RataDie {
```

Comment:
not sure if we really need to cache this but it's easy to compute so probably fine